src/api/archive_unstable_storage.md

```diff
@@ -3,91 +3,129 @@
 
-While the JSON-RPC server is waiting for `archive_unstable_storageContinue` to be called, it can generate a `stop` event indicating that it can no longer proceed with that storage access.
+- `key` is guaranteed to start with one of the `key`s provided as parameter
+- `hash` is a string containing the hexadecimal-encoded value of the storage entry.
 
-### done
 
-```json
+### ClosestDescendantMerkleValue
+
+```
 {
-    "event": "done"
+    "key": "0x0000000...",
+    "closestDescendantMerkleValue": "0x000000..."
 }
 ```
 
-The `done` event indicates that everything went well and all values have been provided through `item` events in the past.
+The JSON object corresponds to one of the requested items whose `type` was `"closestDescendantMerkleValue"`.
 
-If no `item` event was yielded, then the storage doesn't contain a value at the given key.
+- `key` is guaranteed to be equal to one of the `key`s provided as parameter
+- `closestDescendantMerkleValue`: the closest trie Merkle value of the `key`
 
-No more event will be generated with this `subscription`.
+If `key` doesn't exist in the trie, then the Merkle value of the closest descendant of `key` (including branch nodes) is provided. If `key` doesn't have any descendant in the trie, then the result will not contain any relevant item.
 
-### stop
+The trie node whose Merkle value is indicated in `closestDescendantMerkleValue` is not indicated, as determining the key of this node might incur an overhead for the JSON-RPC server. The Merkle value is equal to either the node value or the hash of the node value, as defined in the [Polkadot specification](https://spec.polkadot.network/chap-state#defn-merkle-value).
 
-```json
-{
-    "event": "stop"
-}
-```
+## Overview
+
+For each item in `items`, the JSON-RPC server must start obtaining the value of the entry with the given `key` from the storage, either from the main trie or from `childTrie`. If `type` is `descendantsValues` or `descendantsHashes`, then it must also obtain the values of all the descendants of the entry.
+
+For the purpose of storage requests, the trie root hash of the child tries of the storage can be found in the main trie at keys starting the bytes of the ASCII string `:child_storage:`. This behaviour is consistent with all the other storage-request-alike mechanisms of Polkadot and Substrate-based chains, such as host functions or libp2p network requests.
+
+This function should be used when the target block is older than the blocks reported by `chainHead_unstable_follow`.
+Use `chainHead_unstable_storage` if instead you want to retrieve the storage of a block obtained by the `chainHead_unstable_follow`.
+
+If the `type` of an item is `value`, and `key` is associated with a storage value in the trie, then the result will include an item that contains this storage value. If `key` is not associated with a storage value in the trie, then the result will not include such item.
```

Comment:
These "If the type..." paragraphs here and below feel like a copy of the description of the possible formats above; maybe they can be pruned, and any extra information they might contain be moved to above? Or vice versa.

Reply:
Have moved parts of the overview above, thanks!